packages/kit/src/exports/vite/preview/index.js

```diff
@@ -71,7 +71,19 @@ export async function preview(vite, vite_config, svelte_config) {
 
 		vite.middlewares.use((req, res, next) => {
 			const original_url = /** @type {string} */ (req.url);
-			const { pathname } = new URL(original_url, 'http://dummy');
+			const { pathname, search } = new URL(original_url, 'http://dummy');
+
+			// if `paths.base === '/a/b/c`, then the root route is `/a/b/c/`,
```

Comment:
Can we add an explanation of why that is? It seems quite surprising to me that we would ignore the trailing slash option for the root page when base is set. I remember putting a lot of effort into an earlier version of Vite to allow it to work without a trailing slash and I don't understand why we'd have this new behavior

Reply:
The original explanation is in this previous PR https://github.com/sveltejs/kit/pull/9343 However it was not applied in vite preview, hence this PR. But I think this got removed in 2.0 because vite always does it? I have to check

Reply:
Oops, sorry, I didn't mean to comment on a closed PR. I was on mobile and thought I was looking at https://github.com/sveltejs/kit/pull/11357